controllers/machine_controller.go

```diff
@@ -350,7 +358,15 @@ func (r *MachineReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 	// We only delete the node after the underlying infrastructure is gone.
 	// https://github.com/kubernetes-sigs/cluster-api/issues/2565
 	if isDeleteNodeAllowed {
+		patchHelper, err := patch.NewHelper(m, r.Client)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
 		logger.Info("Deleting node", "node", m.Status.NodeRef.Name)
+		conditions.MarkFalse(m, clusterv1.MachineNodeHealthyCondition, clusterv1.DeletingReason, clusterv1.ConditionSeverityInfo, "")
```

Comment:
Only concern is early signaling a node as deleting that is operational if infrastructure fails to get deleted; but this will unlikely to happen, so changed.